src/batSeq.mlv

```diff
@@ -138,6 +138,31 @@ let of_list l =
   in
   aux l
 
+let empty = nil
+
+(*$T empty
+   length empty = 0
+ *)
+
+let return x =
+    cons x empty
+
+(*$T return
+  length (return 123) = 1
+  at (return 123) 0 = 123
+  equal (return 123) (of_list [123])
+ *)
+
+let rec unfold f u =
+  match f u with
+  | Some(v, r) -> (fun _ -> Cons(v, unfold f r))
```

Comment:
`fun () ->` (whenever you can use either `_` or `()`, always prefer `()`, because sometimes it will tell you about a type mistake you are making.)

Reply:
I fixed it.